sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -778,6 +780,22 @@ private[hive] class HiveClientImpl(
   }
 
 
+
+}
+
+private[hive] object HiveClientImpl {
+  private lazy val shimDefault = IsolatedClientLoader.hiveVersion(
+    HiveUtils.hiveExecutionVersion) match {
+      case hive.v12 => new Shim_v0_12()
+      case hive.v13 => new Shim_v0_13()
+      case hive.v14 => new Shim_v0_14()
+      case hive.v1_0 => new Shim_v1_0()
+      case hive.v1_1 => new Shim_v1_1()
+      case hive.v1_2 => new Shim_v1_2()
+  }
+
+  private lazy val hiveConf = new HiveConf(classOf[SessionState])
```

Comment:
We still need to pass conf in `class HiveClientImpl` for interacting with metastore, because we need to set it.

I think we do not need it for Hive execution, unless we hit any error.